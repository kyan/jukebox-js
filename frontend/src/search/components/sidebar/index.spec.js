import React from 'react'
import { mount } from 'enzyme'
import Search from './index'

describe('Search', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('render', () => {
    const onCloseMock = jest.fn().mockName('onCloseMock')
    const onSubmitMock = jest.fn().mockName('onSubmitMock')
    const onQueryChangeMock = jest.fn().mockName('onQueryChangeMock')
    const onAddTrackMock = jest.fn().mockName('onAddTrackMock')
    const onPageChangeMock = jest.fn().mockName('onPageChangeMock')

    describe('valid props', () => {
      const track1 = {
        track: {
          name: 'Track name 1',
          uri: 'track1',
          artist: {
            name: 'Artist name 1'
          },
          album: {
            name: 'Album name 1'
          },
          image: 'image1'
        }
      }
      const track2 = {
        track: {
          explicit: true,
          name: 'Track name 2',
          uri: 'track2',
          artist: {
            name: 'Artist name 2'
          },
          album: {
            name: 'Album name 2'
          },
          image: 'image2'
        }
      }
      const wrapper = mount(
        <Search
          onClose={onCloseMock}
          onSubmit={onSubmitMock}
          onQueryChange={onQueryChangeMock}
          onAddTrack={onAddTrackMock}
          onPageChange={onPageChangeMock}
          results={[track1, track2]}
          totalPages={2}
          visible
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('focuses the search input onShow', () => {
        const input = wrapper.find('input')
        const sidebar = wrapper.find('Sidebar')
        sidebar.prop('onShow')()
        expect(input.get(0).ref.current).toEqual(document.activeElement)
      })

      it('adds a track', () => {
        const track = wrapper.find('.search-list-item').first()
        track.simulate('click')
        expect(onAddTrackMock).toHaveBeenCalledWith('track1')
      })

      it('closes the sidebar', () => {
        const sidebar = wrapper.find('SidebarPusher')
        sidebar.simulate('click')
        expect(onCloseMock).toHaveBeenCalled()
      })

      it('ignores closing the sidebar', () => {
        const wrapper = mount(
          <Search
            onClose={onCloseMock}
            onSubmit={onSubmitMock}
            onQueryChange={onQueryChangeMock}
            onAddTrack={onAddTrackMock}
            onPageChange={onPageChangeMock}
            results={[track1, track2]}
            totalPages={2}
            visible={false}
          />
        )
        const sidebar = wrapper.find('SidebarPusher')
        sidebar.simulate('click')
        expect(onCloseMock).not.toHaveBeenCalled()
      })
    })
  })
})
