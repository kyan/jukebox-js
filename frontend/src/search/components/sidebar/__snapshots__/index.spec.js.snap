// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render valid props renders as expected 1`] = `
<Search
  onAddTrack={[MockFunction onAddTrackMock]}
  onClose={[MockFunction onCloseMock]}
  onPageChange={[MockFunction onPageChangeMock]}
  onQueryChange={[MockFunction onQueryChangeMock]}
  onSubmit={[MockFunction onSubmitMock]}
  results={
    Array [
      Object {
        "track": Object {
          "album": Object {
            "name": "Album name 1",
          },
          "artist": Object {
            "name": "Artist name 1",
          },
          "image": "image1",
          "name": "Track name 1",
          "uri": "track1",
        },
      },
      Object {
        "track": Object {
          "album": Object {
            "name": "Album name 2",
          },
          "artist": Object {
            "name": "Artist name 2",
          },
          "image": "image2",
          "name": "Track name 2",
          "uri": "track2",
        },
      },
    ]
  }
  totalPages={2}
  visible={true}
>
  <SidebarPushable>
    <div
      className="pushable"
    >
      <Sidebar
        animation="overlay"
        className="sidebar-search"
        direction="right"
        icon="labeled"
        inverted="true"
        onShow={[Function]}
        target={
          Object {
            "current": Document {
              "location": Location {
                "assign": [Function],
                "hash": "",
                "host": "localhost",
                "hostname": "localhost",
                "href": "http://localhost/",
                "origin": "http://localhost",
                "pathname": "/",
                "port": "",
                "protocol": "http:",
                "reload": [Function],
                "replace": [Function],
                "search": "",
                "toString": [Function],
              },
            },
          }
        }
        vertical="true"
        visible={true}
        width="very wide"
      >
        <Ref
          innerRef={
            Object {
              "current": <div
                class="ui overlay right very wide visible sidebar sidebar-search"
                icon="labeled"
                inverted="true"
                vertical="true"
              >
                <form
                  class="ui inverted form"
                >
                  <div
                    class="field"
                  >
                    <label
                      required=""
                    >
                      SEARCH
                    </label>
                    <input
                      placeholder="track, artist or album"
                    />
                  </div>
                  <button
                    class="ui fluid button"
                    type="submit"
                  >
                    Submit
                  </button>
                </form>
                <div
                  class="ui horizontal divider"
                >
                  <h4
                    class="ui inverted header"
                  >
                    Results
                  </h4>
                </div>
                <div
                  aria-label="Pagination Navigation"
                  class="ui pagination pointing secondary search-list-pagination menu"
                  role="navigation"
                >
                  <a
                    aria-current="false"
                    aria-disabled="false"
                    aria-label="Previous item"
                    class="item"
                    tabindex="0"
                    type="prevItem"
                    value="1"
                  >
                    ⟨
                  </a>
                  <a
                    aria-current="true"
                    aria-disabled="false"
                    class="active item"
                    tabindex="0"
                    type="pageItem"
                    value="1"
                  >
                    1
                  </a>
                  <a
                    aria-current="false"
                    aria-disabled="false"
                    class="item"
                    tabindex="0"
                    type="pageItem"
                    value="2"
                  >
                    2
                  </a>
                  <a
                    aria-current="false"
                    aria-disabled="false"
                    aria-label="Next item"
                    class="item"
                    tabindex="0"
                    type="nextItem"
                    value="2"
                  >
                    ⟩
                  </a>
                </div>
                <div
                  class="ui tiny divided inverted relaxed list"
                  role="list"
                >
                  <div
                    class="search-list-item"
                  >
                    <img
                      class="ui mini left floated image search-list-item__image"
                      src="image1"
                      title="Click to add - Track name 1 - Artist name 1"
                    />
                    <div
                      class="content"
                    >
                      <div
                        class="search-list-item__header"
                      >
                        Track name 1
                         - 
                        Artist name 1
                      </div>
                      <div
                        class="search-list-item__content"
                      >
                        Album name 1
                      </div>
                    </div>
                  </div>
                  <div
                    class="search-list-item"
                  >
                    <img
                      class="ui mini left floated image search-list-item__image"
                      src="image2"
                      title="Click to add - Track name 2 - Artist name 2"
                    />
                    <div
                      class="content"
                    >
                      <div
                        class="search-list-item__header"
                      >
                        Track name 2
                         - 
                        Artist name 2
                      </div>
                      <div
                        class="search-list-item__content"
                      >
                        Album name 2
                      </div>
                    </div>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <div
                  class="ui overlay right very wide visible sidebar sidebar-search"
                  icon="labeled"
                  inverted="true"
                  vertical="true"
                >
                  <form
                    class="ui inverted form"
                  >
                    <div
                      class="field"
                    >
                      <label
                        required=""
                      >
                        SEARCH
                      </label>
                      <input
                        placeholder="track, artist or album"
                      />
                    </div>
                    <button
                      class="ui fluid button"
                      type="submit"
                    >
                      Submit
                    </button>
                  </form>
                  <div
                    class="ui horizontal divider"
                  >
                    <h4
                      class="ui inverted header"
                    >
                      Results
                    </h4>
                  </div>
                  <div
                    aria-label="Pagination Navigation"
                    class="ui pagination pointing secondary search-list-pagination menu"
                    role="navigation"
                  >
                    <a
                      aria-current="false"
                      aria-disabled="false"
                      aria-label="Previous item"
                      class="item"
                      tabindex="0"
                      type="prevItem"
                      value="1"
                    >
                      ⟨
                    </a>
                    <a
                      aria-current="true"
                      aria-disabled="false"
                      class="active item"
                      tabindex="0"
                      type="pageItem"
                      value="1"
                    >
                      1
                    </a>
                    <a
                      aria-current="false"
                      aria-disabled="false"
                      class="item"
                      tabindex="0"
                      type="pageItem"
                      value="2"
                    >
                      2
                    </a>
                    <a
                      aria-current="false"
                      aria-disabled="false"
                      aria-label="Next item"
                      class="item"
                      tabindex="0"
                      type="nextItem"
                      value="2"
                    >
                      ⟩
                    </a>
                  </div>
                  <div
                    class="ui tiny divided inverted relaxed list"
                    role="list"
                  >
                    <div
                      class="search-list-item"
                    >
                      <img
                        class="ui mini left floated image search-list-item__image"
                        src="image1"
                        title="Click to add - Track name 1 - Artist name 1"
                      />
                      <div
                        class="content"
                      >
                        <div
                          class="search-list-item__header"
                        >
                          Track name 1
                           - 
                          Artist name 1
                        </div>
                        <div
                          class="search-list-item__content"
                        >
                          Album name 1
                        </div>
                      </div>
                    </div>
                    <div
                      class="search-list-item"
                    >
                      <img
                        class="ui mini left floated image search-list-item__image"
                        src="image2"
                        title="Click to add - Track name 2 - Artist name 2"
                      />
                      <div
                        class="content"
                      >
                        <div
                          class="search-list-item__header"
                        >
                          Track name 2
                           - 
                          Artist name 2
                        </div>
                        <div
                          class="search-list-item__content"
                        >
                          Album name 2
                        </div>
                      </div>
                    </div>
                  </div>
                </div>,
              }
            }
          >
            <div
              className="ui overlay right very wide visible sidebar sidebar-search"
              icon="labeled"
              inverted="true"
              vertical="true"
            >
              <Form
                as="form"
                inverted={true}
                onSubmit={[MockFunction onSubmitMock]}
              >
                <form
                  className="ui inverted form"
                  onSubmit={[Function]}
                >
                  <FormField>
                    <div
                      className="field"
                    >
                      <label
                        required={true}
                      >
                        SEARCH
                      </label>
                      <input
                        onChange={[MockFunction onQueryChangeMock]}
                        placeholder="track, artist or album"
                      />
                    </div>
                  </FormField>
                  <Button
                    as="button"
                    fluid={true}
                    type="submit"
                  >
                    <Ref
                      innerRef={
                        Object {
                          "current": <button
                            class="ui fluid button"
                            type="submit"
                          >
                            Submit
                          </button>,
                        }
                      }
                    >
                      <RefFindNode
                        innerRef={
                          Object {
                            "current": <button
                              class="ui fluid button"
                              type="submit"
                            >
                              Submit
                            </button>,
                          }
                        }
                      >
                        <button
                          className="ui fluid button"
                          onClick={[Function]}
                          type="submit"
                        >
                          Submit
                        </button>
                      </RefFindNode>
                    </Ref>
                  </Button>
                </form>
              </Form>
              <Divider
                horizontal={true}
              >
                <div
                  className="ui horizontal divider"
                >
                  <Header
                    as="h4"
                    inverted={true}
                  >
                    <h4
                      className="ui inverted header"
                    >
                      Results
                    </h4>
                  </Header>
                </div>
              </Divider>
              <Pagination
                aria-label="Pagination Navigation"
                boundaryRange={1}
                className="search-list-pagination"
                defaultActivePage={1}
                ellipsisItem="..."
                firstItem={null}
                lastItem={null}
                nextItem={
                  Object {
                    "aria-label": "Next item",
                    "content": "⟩",
                  }
                }
                onPageChange={[MockFunction onPageChangeMock]}
                pageItem={Object {}}
                pointing={true}
                prevItem={
                  Object {
                    "aria-label": "Previous item",
                    "content": "⟨",
                  }
                }
                secondary={true}
                siblingRange={1}
                totalPages={2}
              >
                <Menu
                  aria-label="Pagination Navigation"
                  className="search-list-pagination"
                  pagination={true}
                  pointing={true}
                  role="navigation"
                  secondary={true}
                >
                  <div
                    aria-label="Pagination Navigation"
                    className="ui pagination pointing secondary search-list-pagination menu"
                    role="navigation"
                  >
                    <PaginationItem
                      active={false}
                      aria-label="Previous item"
                      content="⟨"
                      key="prevItem-1"
                      onClick={[Function]}
                      type="prevItem"
                      value={1}
                    >
                      <MenuItem
                        active={false}
                        aria-current={false}
                        aria-disabled={false}
                        aria-label="Previous item"
                        content="⟨"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        tabIndex={0}
                        type="prevItem"
                        value={1}
                      >
                        <a
                          aria-current={false}
                          aria-disabled={false}
                          aria-label="Previous item"
                          className="item"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          tabIndex={0}
                          type="prevItem"
                          value={1}
                        >
                          ⟨
                        </a>
                      </MenuItem>
                    </PaginationItem>
                    <PaginationItem
                      active={true}
                      content={1}
                      key="pageItem-1"
                      onClick={[Function]}
                      type="pageItem"
                      value={1}
                    >
                      <MenuItem
                        active={true}
                        aria-current={true}
                        aria-disabled={false}
                        content={1}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        tabIndex={0}
                        type="pageItem"
                        value={1}
                      >
                        <a
                          aria-current={true}
                          aria-disabled={false}
                          className="active item"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          tabIndex={0}
                          type="pageItem"
                          value={1}
                        >
                          1
                        </a>
                      </MenuItem>
                    </PaginationItem>
                    <PaginationItem
                      active={false}
                      content={2}
                      key="pageItem-2"
                      onClick={[Function]}
                      type="pageItem"
                      value={2}
                    >
                      <MenuItem
                        active={false}
                        aria-current={false}
                        aria-disabled={false}
                        content={2}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        tabIndex={0}
                        type="pageItem"
                        value={2}
                      >
                        <a
                          aria-current={false}
                          aria-disabled={false}
                          className="item"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          tabIndex={0}
                          type="pageItem"
                          value={2}
                        >
                          2
                        </a>
                      </MenuItem>
                    </PaginationItem>
                    <PaginationItem
                      active={false}
                      aria-label="Next item"
                      content="⟩"
                      key="nextItem-2"
                      onClick={[Function]}
                      type="nextItem"
                      value={2}
                    >
                      <MenuItem
                        active={false}
                        aria-current={false}
                        aria-disabled={false}
                        aria-label="Next item"
                        content="⟩"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        tabIndex={0}
                        type="nextItem"
                        value={2}
                      >
                        <a
                          aria-current={false}
                          aria-disabled={false}
                          aria-label="Next item"
                          className="item"
                          onClick={[Function]}
                          onKeyDown={[Function]}
                          tabIndex={0}
                          type="nextItem"
                          value={2}
                        >
                          ⟩
                        </a>
                      </MenuItem>
                    </PaginationItem>
                  </div>
                </Menu>
              </Pagination>
              <List
                divided={true}
                inverted={true}
                relaxed={true}
                size="tiny"
              >
                <div
                  className="ui tiny divided inverted relaxed list"
                  role="list"
                >
                  <SearchItems>
                    <div
                      className="search-list-item"
                      key="track1"
                      onClick={[Function]}
                    >
                      <Image
                        as="img"
                        className="search-list-item__image"
                        floated="left"
                        size="mini"
                        src="image1"
                        title="Click to add - Track name 1 - Artist name 1"
                        ui={true}
                      >
                        <img
                          className="ui mini left floated image search-list-item__image"
                          src="image1"
                          title="Click to add - Track name 1 - Artist name 1"
                        />
                      </Image>
                      <ListContent>
                        <div
                          className="content"
                        >
                          <div
                            className="search-list-item__header"
                          >
                            Track name 1
                             - 
                            Artist name 1
                          </div>
                          <div
                            className="search-list-item__content"
                          >
                            Album name 1
                          </div>
                        </div>
                      </ListContent>
                    </div>
                    <div
                      className="search-list-item"
                      key="track2"
                      onClick={[Function]}
                    >
                      <Image
                        as="img"
                        className="search-list-item__image"
                        floated="left"
                        size="mini"
                        src="image2"
                        title="Click to add - Track name 2 - Artist name 2"
                        ui={true}
                      >
                        <img
                          className="ui mini left floated image search-list-item__image"
                          src="image2"
                          title="Click to add - Track name 2 - Artist name 2"
                        />
                      </Image>
                      <ListContent>
                        <div
                          className="content"
                        >
                          <div
                            className="search-list-item__header"
                          >
                            Track name 2
                             - 
                            Artist name 2
                          </div>
                          <div
                            className="search-list-item__content"
                          >
                            Album name 2
                          </div>
                        </div>
                      </ListContent>
                    </div>
                  </SearchItems>
                </div>
              </List>
              <EventListener
                capture={false}
                listener={[Function]}
                targetRef={
                  Object {
                    "current": Document {
                      "location": Location {
                        "assign": [Function],
                        "hash": "",
                        "host": "localhost",
                        "hostname": "localhost",
                        "href": "http://localhost/",
                        "origin": "http://localhost",
                        "pathname": "/",
                        "port": "",
                        "protocol": "http:",
                        "reload": [Function],
                        "replace": [Function],
                        "search": "",
                        "toString": [Function],
                      },
                    },
                  }
                }
                type="click"
              />
            </div>
          </RefFindNode>
        </Ref>
      </Sidebar>
      <SidebarPusher
        dimmed={true}
        onClick={[MockFunction onCloseMock]}
      >
        <div
          className="pusher dimmed"
          onClick={[MockFunction onCloseMock]}
        />
      </SidebarPusher>
    </div>
  </SidebarPushable>
</Search>
`;
